### PR TITLE
Displays leaderboard rank in account section

### PR DESCRIPTION
**Pull request by YummyOreo**
### Description

Adds your leaderboard rank for 15 and 60 seconds and you highest daily leaderboard rank and what leaderboard it is from

Screenshot 2022-06-19 171929

### CODE REVIEW

**Comment by Bruception** [comment]
This doesn't work with multiple languages

File: frontend/src/ts/pages/account.ts [hunk 1]
```diff
@@ -631,6 +633,97 @@ export function update(): void {
...
 
+    // checks if the account has typed for more than 2 hours
+    if ((DB.getSnapshot().globalStats?.time ?? 0) >= 7200) {
+      eligableToRanking = true;
+    }
+
+    const modeDifferences = ["15", "60"];
+    const mode = "time";
+
+    const rankRequest = modeDifferences.map((mode2) => {
```
Comment (Bruception) [inline frontend/src/ts/pages/account.ts:644]: Why make the requests if they are not eligible?

File: frontend/src/ts/pages/account.ts [hunk 2]
```diff
@@ -259,6 +260,8 @@ export function update(): void {
     let totalAcc = 0;
     let totalAcc10 = 0;
 
+    let eligableToRanking = false;
```
Comment (Bruception) [inline frontend/src/ts/pages/account.ts:263]: Typo?

File: frontend/src/ts/pages/account.ts [hunk 3]
```diff
@@ -631,6 +633,97 @@ export function update(): void {
...
+        mode,
+        mode2,
+        isDaily: false,
+      });
+    });
+
+    Promise.all(rankRequest).then((data) => {
+      for (let i = 0; i < 2; i++) {
+        if (data[i].status !== 200) {
+          $(`.pageAccount .userRank${i} .error`).text("Error retreaving rank");
```
Comment (Bruception) [inline frontend/src/ts/pages/account.ts:656]: Typo

File: frontend/src/ts/pages/account.ts [hunk 4]
```diff
@@ -631,6 +633,97 @@ export function update(): void {
...
+        language: "english",
+        mode,
+        mode2,
+        isDaily: true,
+        daysBefore: 0,
+      });
+    });
+
+    Promise.all(rankRequestDaily).then((data) => {
+      console.log(data);
```
Comment (Bruception) [inline frontend/src/ts/pages/account.ts:696]: Remove console log

File: frontend/src/ts/pages/account.ts [hunk 5]
```diff
@@ -631,14 +633,104 @@ export function update(): void {
...
+      const rankRequest = modeDifferences.map((mode2) => {
+        return Ape.leaderboards.getRank({
+          language: Config.language,
+          mode,
+          mode2,
+          isDaily: false,
+        });
+      });
+
+      Promise.all(rankRequest).then((data) => {
```
Comment (Bruception) [inline frontend/src/ts/pages/account.ts:654]: We prefer to use async/await syntax over promises. Please refactor this.

**Comment by Bruception** [comment]
As discussed previously, this should be worked on after user profiles have been merged to master.

**Comment by YummyOreo** [comment]
> As discussed previously, this should be worked on after user profiles have been merged to master.

Oh ok i forgot

**Pull request closed without merging**